packages/runtime/src/public/createApp.ts

```diff
@@ -0,0 +1,126 @@
+// Note: We should be careful of what we import in this module, we don't want to load the entire runtime here
+// until createApp is called, since we set some globals here, which will be read by other runtime modules
+import '@mobilestack-xyz/runtime/src/missingGlobals'
+import Config from 'react-native-config'
+import DeviceInfo from 'react-native-device-info'
+import { ToggleableOnboardingFeatures } from '../onboarding/types'
+
+// This will evolve. We should be mindful of breaking changes.
+// This structure should scale well as we add more features
+// and makes it clear what's core configuration vs optional features.
+//
+// Guidelines:
+// - We should only have a few core configuration options, and the rest should be optional features and/or with default values
+// - We should only add new configuration options that we want to support long term, and not just for a specific app
+// - Configuration options should be well documented and have clear purposes
+// - Breaking changes to configuration should be avoided when possible
+// - Configuration should be type-safe. In some cases we can consider runtime validation.
+interface PublicAppConfig {
+  registryName: string
+  displayName: string
+  deepLinkUrlScheme: string
+
+  // Platform specific configuration
+  ios?: {
+    appStoreId?: string
+  }
+
+  // Theme configuration
+  themes?: {
+    // Rough example of what we might want to support
+    default: {
+      // To adjust status bar style, keyboard appearance, etc
+      isDark?: boolean
+      colors?: {
+        // Core brand colors
+        primary?: string
+        secondary?: string
+        background?: string
+
+        // Semantic colors
+        error?: string
+        success?: string
+        warning?: string
+
+        // Text colors
+        text?: {
+          primary?: string
+          secondary?: string
+          disabled?: string
+        }
+      }
+    }
+  }
+
+  // Optional features/capabilities
+  features?: {
```

Comment:
This is not meant to be exhaustive at the moment, correct? Or is this just meant to contain non-secret values at the moment, and will be updated to replace `secrets.json` at some later point? Just from a quick scan of the contents of our `config.ts` file, there are a handful of values in there not present here.